shared/src/packages/users/libs/validation-schemas/common-rules/signup-rules.ts

```diff
@@ -0,0 +1,44 @@
+import joi from 'joi';
+
+import {
+  UserValidationMessage as Message,
+  UserValidationRule as Rule,
+} from '../enums/enums.js';
+
+const passwordPattern = new RegExp(Rule.PASSWORD);
+const namePattern = new RegExp(Rule.NAME);
+const phonePattern = new RegExp(Rule.PHONE);
+
+const signUpRules = {
+  phone: joi.string().trim().required().pattern(phonePattern).messages({
```

Comment:
```suggestion
  phone: joi.string().trim().required().pattern(createPatternByRule(rule)).messages({
```
 It's better to add some helper which will return new RegExp(rule), so we can avoid to write each time each variable for new pattern in this file

Reply:
done